weave/trace_server/opentelemetry/attributes.py

```diff
@@ -313,199 +308,90 @@ def unflatten_key_values(
-    OPENINFERENCE = "openinference"
-    OPENTELEMETRY = "gen_ai"
-    CUSTOM = "custom"
-
-
-class Attributes:
-    _attributes: dict[str, Any] = field(default_factory=dict)
-
-    def __init__(self, _attributes: dict[str, Any] = {}) -> None:
-        self._attributes = _attributes
-
-    def __getitem__(self, key: str) -> Any:
-        return self._attributes.__getitem__(key)
-
-    def __setitem__(self, key: str, value: Any) -> None:
-        return self._attributes.__setitem__(key, value)
-
-    def get(self, key: str, default: Any = None) -> Any:
-        return self._attributes.get(key, default)
-
-    def get_attribute_value(self, key: str) -> Any:
-        return get_attribute(self._attributes, key)
-
-    @abstractmethod
-    def get_weave_inputs(self) -> dict[str, Any]: ...
-
-    @abstractmethod
-    def get_weave_outputs(self) -> Any: ...
-
-    @abstractmethod
-    def get_weave_usage(self) -> LLMUsageSchema: ...
-
-    @abstractmethod
-    def get_weave_summary(self) -> SummaryInsertMap: ...
-
-    @abstractmethod
-    def get_weave_attributes(self, extra: Optional[dict[str, Any]] = None) -> Any: ...
-
-
-# If we don't have any conventions to follow, just dump everything to attributes
-# Later we may add support for user defined conventions through headers
-class GenericAttributes(Attributes):
-    def get_weave_usage(self) -> LLMUsageSchema:
-        return {}
-
-    def get_weave_summary(self) -> SummaryInsertMap:
-        return {}
+    return expand_attributes(iterator)
 
-    def get_weave_inputs(self) -> Any:
-        return {}
 
-    def get_weave_outputs(self) -> Any:
-        return {}
+def try_parse_json(value: Any) -> Any:
+    if isinstance(value, str):
+        try:
+            value = json.loads(value)
+        except:
+            pass
+    return value
 
-    def get_weave_attributes(self, extra: Optional[dict[str, Any]] = None) -> Any:
-        result = self._attributes.copy()
-        result.update(extra or {})
-        return result
 
+def try_parse_int(value: Any) -> Any:
+    try:
+        value = int(value)
+    except:
+        pass
+    return value
 
-class OpenInferenceAttributes(Attributes):
-    def get_weave_usage(self) -> LLMUsageSchema:
-        prompt_tokens = self.get_attribute_value(
-            oi.SpanAttributes.LLM_TOKEN_COUNT_PROMPT
-        )
-        completion_tokens = self.get_attribute_value(
-            oi.SpanAttributes.LLM_TOKEN_COUNT_COMPLETION
-        )
-        total_tokens = self.get_attribute_value(oi.SpanAttributes.LLM_TOKEN_COUNT_TOTAL)
-        return LLMUsageSchema(
-            prompt_tokens=int(prompt_tokens) if prompt_tokens else None,
-            completion_tokens=int(completion_tokens) if completion_tokens else None,
-            total_tokens=int(total_tokens) if total_tokens else None,
-        )
 
-    def get_weave_summary(self) -> SummaryInsertMap:
-        return SummaryInsertMap(usage={"usage": self.get_weave_usage()})
+KEY_HANDLERS = {
```

Comment:
for clarity it would be nice to colocate your mappings with the handlers

Reply:
I'll make a followup ticket for this. I definitely agree but right now it will cause a circular dependency to put this in constants.py and it's much clearer to have constants.py isolated.

I'll open a ticket for a followup to move the helper functions which include the handlers used here to `helpers.py`. I'd rather not obscure the functional changes in this diff further by just moving around large blocks of code.

Reply:
I ended up just doing it so no need to worry about follow up